src/util/ConfigManager/ConfigOptionProxy.h

```diff
@@ -27,9 +28,9 @@ access to the referenced config option.
 to. Must be `ConfigOption`, or `const ConfigOption`.
 */
 template <typename T, typename ConfigOptionType>
-requires ad_utility::isTypeContainedIn<T, ConfigOption::AvailableTypes> &&
-         (std::same_as<ConfigOptionType, ConfigOption> ||
-          std::same_as<ConfigOptionType, const ConfigOption>)
+requires ad_utility::SameAsAnyTypeIn<T, ConfigOption::AvailableTypes> &&
+         ad_utility::SameAsAny<ConfigOptionType, ConfigOption,
+                               const ConfigOption>
```

Comment:
Also directly use them as concepts instead of `typenbame` I think you then get the better diagnostics on clang.